pkg/repositories/transformers/signal.go

```diff
@@ -0,0 +1,134 @@
+package transformers
+
+import (
+	"github.com/flyteorg/flyteadmin/pkg/errors"
+	"github.com/flyteorg/flyteadmin/pkg/repositories/models"
+
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/admin"
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/core"
+
+	"github.com/golang/protobuf/proto"
+
+	"google.golang.org/grpc/codes"
+)
+
+func CreateSignalModel(signalID *core.SignalIdentifier, signalType *core.LiteralType, signalValue *core.Literal) (models.Signal, error) {
+	signalModel := models.Signal{}
+	if signalID != nil {
```

Comment:
when would this ever be nil?

Reply:
It should never be, a signal without an ID is invalid. Included it for completeness. Perhaps throwing an error would be better here?

Reply:
sure that sounds good